src/celengine/body.h

```diff
@@ -240,6 +240,9 @@ class Body : public CatEntry
     PlanetarySystem* getSatellites() const;
     void setSatellites(PlanetarySystem*);
 
+    bool hasPrimary() const;
+    CatEntry* getPrimary() const;
```

Comment:
I'm not sure that CatEntry* is a correct type here, I suppose Selection should be used here